### PR TITLE
allow palette footer to be customised

### PR DESCRIPTION
**Pull request by ahgittin**

### CODE REVIEW

File: ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js [hunk 1]
```diff
@@ -235,6 +235,9 @@ function controller($scope, $element, $q, $uibModal, $log, $templateCache, palet
     // downstream can override this to insert lines below the header
     $scope.customSubHeadTemplateName = 'composer-palette-empty-sub-head';
     $templateCache.put($scope.customSubHeadTemplateName, '');
+    
+    $scope.customFooterTemplateName = 'composer-palette-empty-foort';
```
Comment (aledsage) [inline ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js]: Typo? Should this be `empty-footer`?

File: ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.template.html [hunk 2]
```diff
@@ -61,7 +61,7 @@
             <!-- here and below, col-xs-3 or -4 or -2 all work giving different densities;
                  this could be configurable ("compressed"=xs-2 w no labels, "normal"=xs-3, "big"=xs-4) -->
             <div class="col-xs-3 catalog-palette-item"
-                    ng-repeat="item in searchedItems = (filterPaletteItems(items) | catalogSelectorSearch:search | catalogSelectorSort:family) | orderBy:state.currentOrder | limitTo:pagination.itemsPerPage:(pagination.page-1)*pagination.itemsPerPage track by (item.containingBundle + ':' + item.symbolicName + ':' + item.version)"
+                    ng-repeat="item in searchedItems = (filterPaletteItems(items | catalogSelectorSearch:search) | catalogSelectorSort:family) | orderBy:state.currentOrder | limitTo:pagination.itemsPerPage:(pagination.page-1)*pagination.itemsPerPage track by (item.containingBundle + ':' + item.symbolicName + ':' + item.version)"
```
Comment (aledsage) [inline ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.template.html]: Not sure I follow how `filterPaletteItems` is being used here. Can you give a very brief description, for my benefit please?

File: ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js (hunk 1 above)
Comment (ahgittin) [reply ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js]: oops yes

File: ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.template.html (hunk 2 above)
Comment (ahgittin) [reply ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.template.html]: yes, will add as comment

`// filter by search word first, then pass to any third-party filters, so filters can adjust based on number of search resullts`

**Pull request merged**